### PR TITLE
Travis release

### PR DESCRIPTION
**Pull request by rleon**
Create semi-automatic RDMA core release procedure

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,62 +1,40 @@
 language: c
-# We need at least cmake 2.12, this means we need to use trusty.
-# Precise's glibc, etc predates what we are willing to support.
-# sudo is required to get the trusty image, and at present to enable llvm 3.9
 sudo: required
```
Comment (jgunthorpe) [inline .travis.yml]: Please put all the comments back... And maybe undo the whitespace damage..

File: .travis.yml [hunk 2]
```diff
@@ -1,62 +1,40 @@
...
-  - CC=clang-3.9 CFLAGS=-Werror cmake -GNinja ..
-  - ninja
+- buildlib/travis-build
+before_deploy:
+- buildlib/github-release
+deploy:
+  provider: releases
+  api_key:
+    secure: G9sQd578nctgaety6f27P4p5nyrEgXdvMs1g+K2Hj7GPWxBpQ1Jm7kUlahcwfnYXpYqXen9b2dzyGzaLxjLL5rv6rbtOYp9SVrvybQrnJwkB0HCERKVMwEPsLPOB2lMouBDfk4+iUGdxtcw8Y5UU/xqKcAjkHkM5YpF5HauEX1Jj+Ep/XpX5DEcsVmFwkDgPKJOZv/eUgz1eVtXNFIvmSwJ+QuuUz64JpEpKMY4u7bFu0+2qWqACj19P/Rnufxpr+o5yoCuVNVptoRHMIWJiWZsrd2IhmL3knBoPoxNXSOWqQCK6DCW1KYOOVVAfFoD4YWhWrt6MuB3hh7lgLkS0rsdG9+iMlosl4SOkWZF8ZTkmrmtt6yx+s18tFdNI46QKzaA4wS/1hktweIFizYvmEwUCU8sNK5y9PMfDoGJGwiF1FzWrnLItKMiib0sTGO4Umgc0DKTazVRlIJYooOOOnBGubRudNx3elBTy4C/txSmqR39/6GYlvcyUDJMqB/oQJj2erQ5Gu8VykW5zjko6+3miX3C7GFVQ781BLxAdyNl4WRjZ9ibzjMbK3MC7f0ho1hST6cvBBmy2UnIdvB6zMayHZ/gtXKsDqE386k5FwSuTLT2V7942uFRVJiEwhVmgmBzbZND7PTj4s9GPrL2HLlJOgZooRZdx/ZcT/YNPFbA=
+  file: rdma-core-*.tar.gz
```
Comment (jgunthorpe) [inline .travis.yml]: What is this and where did it come from? Why is it safe? Needs a comment..

File: buildlib/travis-build [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+CC=gcc-6 CFLAGS="-Werror -m32" cmake -GNinja .. -DENABLE_RESOLVE_NEIGH=0
+ninja
+
+# Test with coherent DMA mode disabled (ie as would be on ARM32, etc)
+cd ../build-clang
+echo "#error Fail" >> ../libibverbs/arch.h
+rm CMakeCache.txt
+CC=clang-3.9 CFLAGS=-Werror cmake -GNinja ..
+ninja
+cd ..
```
Comment (jgunthorpe) [inline buildlib/travis-build]: I think this causes the travis log to be much harder to read - travis puts special colourised annotations including runtime between each step with the old method..

Maybe at least stick a few echo lines between the steps?

File: .travis.yml (hunk 1 above)
Comment (rleon) [inline .travis.yml]: Sure, I'll return previous version of travis.yml.

File: .travis.yml (hunk 2 above)
Comment (rleon) [inline .travis.yml]: I added comment, it comes from "travis setup releases" command.

File: buildlib/travis-build (hunk 3 above)
Comment (rleon) [inline buildlib/travis-build]: I added set -x at the beginning of this file

**Pull request merged**